### PR TITLE
JobPosting pagination

### PR DESCRIPTION
**Pull request by ReneSG**
What does this PR do?
- Adds pagination to JobPosting index view.
- Adds custom renderer to handle ui for pagination buttons and pages.

image

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -13,4 +13,39 @@ def after_sign_in_path_for(_resource)
...
+  def custom_paginate_renderer
+    # Return nice pagination for materialize
+    Class.new(WillPaginate::ActionView::LinkRenderer) do
+      def container_attributes
+        {class: "pagination"}
+      end
+
+      def page_number(page)
+        if page == current_page
+          "<li class=\"cyan active\">"+link(page, page, rel: rel_value(page))+"</li>"
```
Comment (eduardotru) [inline app/controllers/application_controller.rb:26]: Can you change it to be:
```
"<li class=\"cyan active\">#{link(page, page, rel: rel_value(page))}</li>"
```

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -13,4 +13,39 @@ def after_sign_in_path_for(_resource)
...
+        else
+          "<li class=\"waves-effect\">"+link(page, page, rel: rel_value(page))+"</li>"
```
Comment (eduardotru) [inline app/controllers/application_controller.rb:28]: Ditto

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -13,4 +13,39 @@ def after_sign_in_path_for(_resource)
...
+      end
+
+      def next_page
+        num = @collection.current_page < total_pages && @collection.current_page + 1
+        previous_or_next_page(num, "<i class=\"material-icons\">chevron_right</i>")
+      end
+
+      def previous_or_next_page(page, text)
+        if page
+          "<li class=\"waves-effect\">"+link(text, page)+"</li>"
```
Comment (eduardotru) [inline app/controllers/application_controller.rb:44]: Ditto

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -13,4 +13,39 @@ def after_sign_in_path_for(_resource)
...
+        else
+          "<li class=\"waves-effect\">"+text+"</li>"
```
Comment (eduardotru) [inline app/controllers/application_controller.rb:46]: Ditto

File: app/models/job_posting.rb [hunk 5]
```diff
@@ -1,6 +1,8 @@
 class JobPosting < ApplicationRecord
   belongs_to :company
 
+  self.per_page = 10
```
Comment (eduardotru) [inline app/models/job_posting.rb:4]: Where is this used? Also, if this is the number of opportunities per page we should consider increasing it to a multiple of the columns (if we have 3 columns then it should be either 9 or 12)

Comment (ReneSG) [reply app/models/job_posting.rb:4]: It's used by will_paginate.

**Pull request merged**